build-support/bin/mitol/build_support/commands/changelog.py

```diff
@@ -24,25 +24,29 @@ def changelog(ctx):
     makedirs("changelog.d", exist_ok=True)
 
 
-changelog.add_command(app_option(create))
-changelog.add_command(app_option(collect))
+changelog.add_command(apps_option_no_default(create))
+changelog.add_command(apps_option_no_default(collect))
 
 
 @changelog.command("list")
-@app_option
-@pass_app
+@apps_option
+@pass_apps
 @pass_context
-def list_all(ctx: Context, app: App):
+def list_all(ctx: Context, apps: Apps):
```

Comment:
```
mattbert@Matts-MacBook-Pro ol-django % ./pants ol-project release create --app hubspot_api --push
Traceback (most recent call last):
  File "/Users/mattbert/.pyenv/versions/3.8.14/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/mattbert/.pyenv/versions/3.8.14/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/__main__.py", line 103, in <module>
    bootstrap_pex(__entry_point__, execute=__execute__, venv_dir=__venv_dir__)
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/.bootstrap/pex/pex_bootstrapper.py", line 608, in bootstrap_pex
    pex.PEX(entry_point).execute()
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/.bootstrap/pex/pex.py", line 560, in execute
    sys.exit(self._wrap_coverage(self._wrap_profiling, self._execute))
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/.bootstrap/pex/pex.py", line 467, in _wrap_coverage
    return runner(*args)
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/.bootstrap/pex/pex.py", line 498, in _wrap_profiling
    return runner(*args)
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/.bootstrap/pex/pex.py", line 603, in _execute
    return self.execute_entry(
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/.bootstrap/pex/pex.py", line 798, in execute_entry
    return self.execute_entry_point(entry_point)
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/.bootstrap/pex/pex.py", line 815, in execute_entry_point
    runner = entry_point.resolve()
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/.bootstrap/pex/dist_metadata.py", line 732, in resolve
    module = importlib.import_module(self.module)
  File "/Users/mattbert/.pyenv/versions/3.8.14/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/mitol/build_support/cli.py", line 3, in <module>
    from mitol.build_support.commands.changelog import changelog
  File "/Users/mattbert/.pex/unzipped_pexes/02febc0e99392508e6ec23ed623006b4d0f4dc80/mitol/build_support/commands/changelog.py", line 35, in <module>
    def list_all(ctx: Context, apps: Apps):
TypeError: apps_option() takes 0 positional arguments but 1 was given
```

If I change that function to take optional args:
```python
def apps_option(default=AllApps):
```

then I get this (I think stacked decorators are messing w/each other's args/kwargs?):
```
Traceback (most recent call last):
  File "/Users/mattbert/.pyenv/versions/3.8.14/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/mattbert/.pyenv/versions/3.8.14/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/__main__.py", line 103, in <module>
    bootstrap_pex(__entry_point__, execute=__execute__, venv_dir=__venv_dir__)
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/.bootstrap/pex/pex_bootstrapper.py", line 608, in bootstrap_pex
    pex.PEX(entry_point).execute()
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/.bootstrap/pex/pex.py", line 560, in execute
    sys.exit(self._wrap_coverage(self._wrap_profiling, self._execute))
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/.bootstrap/pex/pex.py", line 467, in _wrap_coverage
    return runner(*args)
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/.bootstrap/pex/pex.py", line 498, in _wrap_profiling
    return runner(*args)
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/.bootstrap/pex/pex.py", line 603, in _execute
    return self.execute_entry(
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/.bootstrap/pex/pex.py", line 798, in execute_entry
    return self.execute_entry_point(entry_point)
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/.bootstrap/pex/pex.py", line 815, in execute_entry_point
    runner = entry_point.resolve()
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/.bootstrap/pex/dist_metadata.py", line 732, in resolve
    module = importlib.import_module(self.module)
  File "/Users/mattbert/.pyenv/versions/3.8.14/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/mitol/build_support/cli.py", line 4, in <module>
    from mitol.build_support.commands.release import release
  File "/Users/mattbert/.pex/unzipped_pexes/664364fb13392b66404557d52e7b69969d1b20ed/mitol/build_support/commands/release.py", line 78, in <module>
    def push_to_remote(project: Project, apps: Apps):
  File "/Users/mattbert/.pex/installed_wheels/d11a8eb7526ffd0fce3650e7f3308260970a87d23f1f07b77ac2f40b5ce672d2/cloup-2.0.0.post1-py2.py3-none-any.whl/cloup/_commands.py", line 303, in decorator
    self.add_command(cmd, section=section)
  File "/Users/mattbert/.pex/installed_wheels/d11a8eb7526ffd0fce3650e7f3308260970a87d23f1f07b77ac2f40b5ce672d2/cloup-2.0.0.post1-py2.py3-none-any.whl/cloup/_commands.py", line 159, in add_command
    super().add_command(cmd, name, section, fallback_to_default_section)
  File "/Users/mattbert/.pex/installed_wheels/d11a8eb7526ffd0fce3650e7f3308260970a87d23f1f07b77ac2f40b5ce672d2/cloup-2.0.0.post1-py2.py3-none-any.whl/cloup/_sections.py", line 191, in add_command
    self._add_command_to_section(cmd, name, section)
  File "/Users/mattbert/.pex/installed_wheels/d11a8eb7526ffd0fce3650e7f3308260970a87d23f1f07b77ac2f40b5ce672d2/cloup-2.0.0.post1-py2.py3-none-any.whl/cloup/_sections.py", line 142, in _add_command_to_section
    section.add_command(cmd, name)
  File "/Users/mattbert/.pex/installed_wheels/d11a8eb7526ffd0fce3650e7f3308260970a87d23f1f07b77ac2f40b5ce672d2/cloup-2.0.0.post1-py2.py3-none-any.whl/cloup/_sections.py", line 62, in add_command
    raise Exception(f'command "{name}" already exists')
Exception: command "-apps-option" already exists
```